lib/document_provider.ex

```diff
@@ -104,8 +110,7 @@ defmodule AbsintheCache.DocumentProvider do
               # This can lead to infinite storing the same value
               Process.put(:do_not_cache_query, true)
 
-              {:jump, %{bp_root | result: result},
-               AbsintheCache.Phase.Document.Execution.Idempotent}
+              {:jump, %{bp_root | result: result}, AbsintheCache.DocumentProvider.Idempotent}
```

Comment:
When the module definition place was changed, this affected the whole module name.
Before that, it was mirroring the Absinthe path like: `Absinthe.Phase.Document.Execution.Resolution` and `Absinthe.Phase.Document.Result`.

I think it would be best to define these modules in another directory `phases` that holds the custom phases defined as modules. This way it would be also much clearer how to refer to this phase from within or outside the module. Currently `CacheDocument` and `Idempotent` will have very different paths.

Reply:
Due to inactivity, I'll merge this change and submit a follow-up PR with my suggested changes.